### PR TITLE
Rename Person to Passenger, Human to Person

### LINKED ISSUE #58: Rename Person to Passenger, Human to Person
Refactor Person to Passenger, Human to Person, to improve from stubbing

### PR DESCRIPTION
**Pull request by JoelHo**

### CODE REVIEW

File: src/main/java/seedu/address/logic/Logic.java [hunk 1]
```diff
@@ -30,11 +30,15 @@
      */
     ReadOnlyAddressBook getAddressBook();
 
-    /** Returns an unmodifiable view of the filtered list of persons */
-    ObservableList<Person> getFilteredPersonList();
+    /**
+     * Returns an unmodifiable view of the filtered list of passengers.
```
Comment (lyueyang) [inline src/main/java/seedu/address/logic/Logic.java:34]: Should the javadocs be represented like this or would a single line as per the original document be better?

File: src/main/java/seedu/address/logic/Logic.java [hunk 2]
```diff
@@ -30,11 +30,15 @@
...
+     **/
+    ObservableList<Passenger> getFilteredPassengerList();
 
-    /** Returns an unmodifiable view of the filtered list of persons whether it is assigned to a driver already*/
-    ObservableList<Person> getFilteredPersonListByDriverStatus(Boolean isAssigned);
+    /**
+     * Returns an unmodifiable view of the filtered list of passengers whether it is assigned to a driver already.
```
Comment (lyueyang) [inline src/main/java/seedu/address/logic/Logic.java:39]: same goes for here as per above

File: src/main/java/seedu/address/storage/JsonAdaptedPassenger.java [hunk 3]
```diff
@@ -54,13 +54,13 @@ public JsonAdaptedPerson(@JsonProperty("name") String name, @JsonProperty("phone
...
     /**
-     * Converts a given {@code Person} into this class for Jackson use.
+     * Converts a given {@code Passenger} into this class for Jackson use.
      */
-    public JsonAdaptedPerson(Person source) {
+    public JsonAdaptedPassenger(Passenger source) {
         name = source.getName().fullName;
         phone = source.getPhone().value;
         address = source.getAddress().value;
-        tripDay = source.getTripDay().value.toString();
```
Comment (ZechariahTan) [inline src/main/java/seedu/address/storage/JsonAdaptedPassenger.java:63]: Is the toString method call required? if this Pr is just for renaming was removing the call a mistake?

File: src/main/java/seedu/address/logic/Logic.java (hunk 1 above)
Comment (lyueyang) [reply src/main/java/seedu/address/logic/Logic.java:34]: closing as upon discussion, we agreed that it's alright as checkstyle tests have passed it

File: src/main/java/seedu/address/logic/Logic.java (hunk 2 above)
Comment (lyueyang) [reply src/main/java/seedu/address/logic/Logic.java:39]: closing as upon discussion, we agreed that it's alright as checkstyle tests have passed it

File: src/test/java/seedu/address/testutil/DriverUtil.java [hunk 4]
```diff
@@ -21,10 +21,10 @@ public static String getDriveCommand(Driver driver, Set<Index> indices) {
         return DriveCommand.COMMAND_WORD + " " + getDriverDetails(driver) + getCommuterDetails(indices);
```
Comment (chewterence) [inline src/test/java/seedu/address/testutil/DriverUtil.java:21]: what is the difference between passenger and commuter?

File: src/test/java/seedu/address/storage/JsonAdaptedPassengerTest.java [hunk 5]
```diff
@@ -37,77 +37,77 @@
 
     @Test
     public void toModelType_validPersonDetails_returnsPerson() throws Exception {
```
Comment (chewterence) [inline src/test/java/seedu/address/storage/JsonAdaptedPassengerTest.java]: Is this method meant to be person?

**Review by chewterence — COMMENTED** [review]
apart from the potential minor error i highlighted, LGTM

File: src/test/java/seedu/address/storage/JsonAdaptedPassengerTest.java (hunk 5 above)
Comment (JoelHo) [reply src/test/java/seedu/address/storage/JsonAdaptedPassengerTest.java]: Fixed in commit 1c344c7.

**Comment by JoelHo** [comment]
Closes #58.

**Pull request merged**